src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs

```diff
@@ -46,23 +48,44 @@ public class RequestTrackingMiddleware
         /// <param name="httpContext">The current HTTP context.</param>
         public async Task Invoke(HttpContext httpContext)
         {
-            var stopwatch = Stopwatch.StartNew();
+            Guard.NotNull(httpContext, nameof(httpContext), "Requires a HTTP context to track the request");
+            Guard.NotNull(httpContext.Request, nameof(httpContext), "Requires a HTTP request in the context to track the request");
+            Guard.NotNull(httpContext.Response, nameof(httpContext), "Requires a HTTP response in the context to track the request");
 
-            string requestBody = null;
-            if (_options.IncludeRequestBody)
+            var endpoint = httpContext.Features.Get<IEndpointFeature>();
+            if (endpoint is null)
             {
-                httpContext.Request.EnableBuffering();
-                requestBody = await GetRequestBodyAsync(httpContext);
+                _logger.LogWarning(
```

Comment:
I'd go with a trace here personally - It's not a security risk & still important to have observability, but not worth the tons of telemetry it will generate
```suggestion
                _logger.LogTrace(
```